packages/oats/src/util.ts

```diff
@@ -1,9 +1,48 @@
 import * as oas from 'openapi3-ts';
 import * as assert from 'assert';
 import * as _ from 'lodash';
+import type { Driver } from './driver';
+import { server } from '@smartlyio/oats-runtime';
 
 export type SchemaObject = oas.ReferenceObject | oas.SchemaObject;
 
+export function filterEndpointsInSpec(
+  spec: oas.OpenAPIObject,
+  { includeEndpoints }: Pick<Driver, 'includeEndpoints'>
```

Comment:
allowing this to be a function would allow a bit more ergonomics in filtering

Reply:
we can support an additional function type, for example:
```typescript
export interface Driver {
  includeEndpoints?:
    | Record<string, Uppercase<serverRuntime.Methods>[]>
    | (path: string, method: Uppercase<serverRuntime.Methods>) => boolean;
}
```
if/when there's a need. For now, it's quite handy to have this as a simple record:
first, it's declarative
second, if you by mistake included a non-existing endpoint, you will get an error

Reply:
anyways, i did some refactoring like you suggested. So the function type is supported (not that I see use case for it so far).